web/src/components/auth/signup/AuthSignupVerification.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+import { Text } from "@tremor/react";
+import { AuthSignupRequestVerificationButton } from "@/components/auth/signup/AuthSignupRequestVerificationButton";
+import { User } from "@/lib/types";
+
+export function EmailVerificationComponent({ user }: { user: User | null }) {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+
+  const [error, setError] = useState("");
+
+  async function verify() {
+    const token = searchParams.get("token");
+    if (!token) {
+      setError(
+        "Missing verification token. Try requesting a new verification email."
+      );
+      return;
+    }
+
+    const response = await fetch("/api/auth/verify", {
```

Comment:
Maybe use `axios` for fetches? Thus it will be shorter and we can have interceptors in case we want some headers always be in requests or some token

Reply:
I don't think that is good idea to do it only there. We can make separate issue for whole project